samples/mqtt5/mqtt5_pubsub/README.md

```diff
@@ -69,3 +69,233 @@ You can also pass a Certificate Authority file (CA) if your certificate and key
+    std::shared_ptr<Aws::Crt::Auth::ICredentialsProvider> provider = Aws::Crt::Auth::CredentialsProvider::CreateCredentialsProviderChainDefault(defaultConfig);
+    if (!provider)
+    {
+        fprintf(stderr, "Failure to create credentials provider!\n");
+        exit(-1);
+    }
+    Aws::Iot::WebsocketConfig websocketConfig(<signing region>, provider);
+
+    // Create a Client using Mqtt5ClientBuilder
+    Aws::Iot::Mqtt5ClientBuilder *builder = Aws::Iot::Mqtt5ClientBuilder::NewMqtt5ClientBuilderWithWebsocket(
+        "<clientEndpoint>", websocketConfig);
+
+    /* You can setup other client options and lifecycle event callbacks before call builder->Build().
+    ** Once the the client get built, you could no longer update the client options or connection options
+    ** on the created client.
+    */
+
+    // Build Mqtt5Client
+    std::shared_ptr<Aws::Crt::Mqtt5Client> mqtt5Client = builder->Build();
+
+    if (mqtt5Client == nullptr)
+    {
+        fprintf(stdout, "Client creation failed.\n");
+        return -1;
+    }
+
+
+```
+
+
+### Direct MQTT with Custom Authentication
+
+AWS IoT Core Custom Authentication allows you to use a lambda to gate access to IoT Core resources.  For this authentication method,you must supply an additional configuration structure containing fields relevant to AWS IoT Core Custom Authentication.
+
+
+
+If your custom authenticator does not use signing, you don't specify anything related to the token signature:
+
+```cpp
+    // Setup custom authorization config
+    Mqtt5CustomAuthConfig customAuth;
+    customAuth.WithAuthrizaerName("<Name of your custom authorizer>");
+    customAuth.WithUsername("<Value of the username field that should be passed to the authorizer's lambda>");
+    customAuth.WithPassword(<Binary data value of the password field to be passed to the authorizer lambda>);
+
+    // Create a Client using Mqtt5ClientBuilder
+    Aws::Iot::Mqtt5ClientBuilder *builder = Aws::Iot::Mqtt5ClientBuilder::NewMqtt5ClientBuilderWithCustomCustomAuthorizer(
+        "<clientEndpoint>", customAuth);
+
+    /* You can setup other client options and lifecycle event callbacks before call builder->Build().
+    ** Once the the client get built, you could no longer update the client options or connection options
+    ** on the created client.
+    */
+
+    // Build Mqtt5Client
+    std::shared_ptr<Aws::Crt::Mqtt5Client> mqtt5Client = builder->Build();
+
+    if (mqtt5Client == nullptr)
+    {
+        fprintf(stdout, "Client creation failed.\n");
+        return -1;
+    }
+
+```
+
+
+If your custom authorizer uses signing, you must specify the three signed token properties as well.  The token signature must be the URI-encoding of the base64 encoding of the digital signature of the token value via the private key associated with the public key that was registered with the custom authorizer.  It is your responsibility to URI-encode the token signature.
```

Comment:
> The token signature must be the URI-encoding

Only token name is required to be URI-encoded, the signature will be encoded automatically.

Reply:
Do we remove the entire back two sentences about the token signature then? Also, should we add that the user name must be URI-encoded?

Reply:
Yes, since our code snippet uses username and AuthorizerName, we should say that they must be URI-encoded. As for tokenKeyName -  probably this will be excess, though I'm not sure